components/eamxx/scripts/check-hashes-ers

```diff
@@ -0,0 +1,171 @@
+###############################################################################
+    return hash_ln.split()[1:3]
+
+###############################################################################
+def all_equal(t1, t2):
+###############################################################################
+    if len(t1) != len(t2): return False
+    for i in range(len(t1)):
+        if t1[i] != t2[i]: return False
+    return True
+
+###############################################################################
+def find_first_index_at_time(lns, time):
+###############################################################################
+    for i, ln in enumerate(lns):
+        t = parse_time(ln)
+        if all_equal(time, t): return i
+    return None
+
+###############################################################################
+def diff(l1, l2):
+###############################################################################
+    diffs = []
+    for i in range(len(l1)):
+        if l1[i] != l2[i]:
+            diffs.append((l1[i], l2[i]))
+    return diffs
+
+###############################################################################
+def check_hashes_ers(case_dir):
+###############################################################################
+    case_dir_p = pathlib.Path(case_dir)
+    expect(case_dir_p.is_dir(), f"{case_dir} is not a dir")
+
+    # Look for the two e3sm.log files.
+    glob_pat = get_log_glob_from_atm_modelio(case_dir_p)
+    e3sm_fns = glob.glob(glob_pat)
+    if len(e3sm_fns) == 0:
+        print('Could not find e3sm.log files with glob string {}'.format(glob_pat))
+        return False
+    e3sm_fns.sort()
+    if len(e3sm_fns) == 1:
+        # This is the first run. Exit and wait for the second
+        # run. (POSTRUN_SCRIPT is called after each of the two runs.)
+        print('Exiting on first run.')
+        return True
+    print('Diffing base {} and restart {}'.format(e3sm_fns[0], e3sm_fns[1]))
+
+    # Because of the prefixed 1: and 2: on some systems, we can't just use
+    # zdiff.
+    lns = []
+    for f in e3sm_fns:
+        lns.append(get_hash_lines(f))
+    time = parse_time(lns[1][0])
+    time_idx = find_first_index_at_time(lns[0], time)
+    if time_idx is None:
+        print('Could not find a start time.')
+        return False
+    lns[0] = lns[0][time_idx:]
+    if len(lns[0]) != len(lns[1]):
+        print('Number of hash lines starting at restart time do not agree.')
+        return False
+    diffs = diff(lns[0], lns[1])
+
+    # Flushed prints to e3sm.log can sometimes conflict with other
+    # output. Permit up to 'thr' diffs so we don't fail due to badly printed
+    # lines. This isn't a big loss in checking because an ERS_Ln22 second run
+    # writes > 1000 hash lines, and a true loss of BFBness is nearly certain to
+    # propagate to a large number of subsequent hashes.
+    thr = 5
+    if len(lns[0]) < 100: thr = 0
+
+    ok = True
+    if len(diffs) > thr:
+        print('DIFF')
```

Comment:
In the same spirit of the other comment, we may inform the avg user of what "DIFF" refers to (namely, that the eamxx bfbhash between base/rest did not match). Maybe even print the first thr+1 occurrences? It may help to immediately see how long it took for non bfb to develop, since I'm guessing we'd see a timestamp in the log line?